src/rabbit_channel.erl

```diff
@@ -789,9 +792,16 @@ check_internal_exchange(#exchange{name = Name, internal = true}) ->
 check_internal_exchange(_) ->
     ok.
 
-check_topic_authorisation(#exchange{name = Name, type = topic}, #ch{user = User}, RoutingKey, Permission) ->
+check_topic_authorisation(#exchange{name = Name = #resource{virtual_host = VHost}, type = topic},
+                          #ch{user = User = #user{username = Username}, conn_pid = ConnPid},
+                          RoutingKey,
+                          Permission) ->
     Resource = Name#resource{kind = topic},
-    Context = #{routing_key => RoutingKey},
+    AmqpConnInfo = gen_server:call(ConnPid, {info, [amqp_params]}, ?TIMEOUT_60s),
```

Comment:
@michaelklishin - is there a better way to define the timeout other than a macro? I wanted to be consistent with the timeouts defined in rabbitmq/rabbitmq-erlang-client#86

There is an `?RPC_TIMEOUT` macro in `rabbit_misc.hrl` that expands to ... `infinity` 😐